passport/src/auth-handler.ts

```diff
@@ -77,28 +82,58 @@ form {
 `);
 }
 
-function onCallback(req: express.Request, res: express.Response, next: (err: any) => void) {
-  passport.authenticate('auth0', (err, user, _info) => {
-    if (err) { return next(err); }
-    if (!user) { return res.redirect('/login'); }
-    req.logIn(user, (loginErr) => {
-      if (err) { return next(loginErr); }
-      res.redirect(req.session?.returnTo || '/');
-      delete req.session?.returnTo;
-    });
-  })(req, res, next);
-}
-
-const oauthPage: express.Handler[] = [
-  (req, _res, next) => {
-    if (req.session) {
-      req.session.returnTo = req.query.returnTo || '/';
+// A poor man's vhost module - calls handler if the hostname matches
+function ifMatch(domain: string, handler: express.Handler): express.Handler {
+  return (req, res, next) => {
+    if (domain === '*' || domain === req.hostname) {
+      return handler(req, res, next);
     }
     next();
-  },
-  passport.authenticate('auth0', { scope: 'openid email profile' }),
-  (_req, res) => res.redirect('/'),
-];
+  };
+}
+
+function createOnCallback(domain: string): express.Handler {
+  return (req, res, next) => {
+    passport.authenticate(`auth0-${domain}`, (err, user, _info, _extra) => {
+      if (err) { return next(err); }
+      if (!user) { return res.redirect('/login'); }
+      req.logIn(user, (loginErr) => {
+        if (err) { return next(loginErr); }
```

Comment:
AFAICT this line is never called, because the containing function would have returned at line 98.  Should this be
```suggestion
        if (loginErr) { return next(loginErr); }
```
?

Reply:
Good catch! Added comment - we are not expected to hit this line - since our `serializeUser` function can not throw, but still better to be correct.